### ISSUE TITLE
Implement Restore Revision dialog iOS

### ISSUE THREAD

**Issue by yaelirub** [wordpress-mobile/WordPress-iOS#12141]
For discovery history - wordpress-mobile/WordPress-Android#10008,  wordpress-mobile/WordPress-Android#9812
If the post has no local changes and a remote autosave, we are going to show a dialog to the user.
This issue is for implementing the design for this dialog.
<img width="422" alt="screenshot-2019-05-03-at-09 25 19" src="https://user-images.githubusercontent.com/40213/57301854-3a9c1600-70da-11e9-8b51-caace0fe6d33.png">

## Matching Android Issue:

https://github.com/wordpress-mobile/WordPress-Android/issues/10108

## Known Tasks

- [ ] Add Tracks events as described in https://github.com/wordpress-mobile/WordPress-Android/issues/10444

###UPDATE -  10/21/19
 the acceptance criteria has evolved through a couple of Android issues and pull requests:

 - iOS will use the system alert (pending design approval) for the dialog
- Posts containing autosaves on the post list screen will display the label "You’ve made unsaved changes to this post", this label previously read "Unhandled Auto Save"
- Even after tapping on the “Restore” button, users can still back out of the post without editing - and if they tap edit again they will be presented with the dialog (this will be done via the "Discard Changes" button, any unsaved changes are discarded)

**Comment by shiki** [comment]
@guarani will be working on this.

**Comment by shiki** [comment]
Moved this out of the Offline Posting board to keep the board focused on “posting”.

**Comment by guarani** [comment]
@yaelirub, regarding https://github.com/wordpress-mobile/WordPress-iOS/issues/12141#issue-468866708:
> If the post **has local changes** and a remote autosave, we are going to show a dialog to the user.

[emphasis added]

After reading https://github.com/wordpress-mobile/WordPress-Android/issues/9812#issue-441225245, I'm wondering if the above comment should read:

_"If the post **has NO local changes** and a remote autosave, we are going to show a dialog to the user."_

I think this is what was done in Android:

https://github.com/wordpress-mobile/WordPress-Android/pull/10426/files#diff-bdb6ebf0ab007bf9820664cba5dd351bR462-R464

**Comment by shiki** [comment]
> "If the post has NO local changes and a remote autosave, we are going to show a dialog to the user."

That's correct, @guarani.

**Comment by guarani** [comment]
@yaelirub, the acceptance criteria has evolved through a couple of Android issues and pull requests, so I'm just updating here for completeness:

1. iOS will use the system alert (pending design approval) for the dialog
2. Posts containing autosaves on the post list screen [will display](https://href.li/?https://github.com/wordpress-mobile/WordPress-Android/issues/10462) the label "You’ve made unsaved changes to this post", this label previously read "Unhandled Auto Save"
3. Even after tapping on the “Restore” button, users [can still back out](https://github.com/wordpress-mobile/WordPress-Android/issues/10450) of the post without editing - and if they tap edit again they will be presented with the dialog (this will be done via the "Discard Changes" button, any unsaved changes are discarded)

**Comment by yaelirub** [comment]
Thanks, @guarani :)
Please feel free to update the issue description according to the update.

**Comment by guarani** [comment]
I don't seem to have edit permission on the issue description - is that something you could enable, @yaelirub?

**Comment by guarani** [comment]
I'm cross-posting a prototype for the dialog here for design approval:

![prototype](https://user-images.githubusercontent.com/1898325/67256902-cecb2100-f45f-11e9-9082-7332f92ba777.png)

(originally posted on P2) cc: @osullivanchris @mattmiklic

**Comment by yaelirub** [comment]
@guarani , no worries. I updated the description.

**Comment by osullivanchris** [comment]
@guarani thanks for digging around and finding the other relevant tickets, particularly the Android issue. As we have a number of similar and related scenarios, it takes me some time to recall which is which when I encounter these issues again. However I can see that this is one I already worked through on Android with Jirka. 

I am happy to go ahead with the proposal here, the only change I would ask is to change the button copy to Sentence case. The all caps styling is just an Android/Material thing.

One question I have is what happens if the text on those buttons gets too long? Is it truncated? I can see that the text is smaller in the longer string here, so maybe that's the solution. Is that part of the system component and a common pattern to use, in your opinion?

Thanks!

**Comment by guarani** [comment]
Thanks Chris, the sentence case escaped me - I'll use that. 
I think the text shrinks up to a certain point, but if the text is too long it gets truncated. I can double check this behavior tonight.

**Comment by guarani** [comment]
@mattmiklic good point - it could vary a lot by language. Personally I find the "From another device"/"From this device" to be the most user friendly.
@osullivanchris, shall we go with one of these options on iOS?

**Comment by osullivanchris** [comment]
Ah tricky! For @mattmiklic's second option, I think its hard to come up with good labels. And its tricky because the user has to parse the information in the dialog and then match it back to the options in the buttons.

As I explored in option 2 here, I thought the way to make selection easier was to provide metadata and action next to each other, so as not to need an abstract name. But as we've found its not supported by the native dialog implementation on either iOS or Android and would require too much customisation.

So I would go with the simpler suggestion from @mattmiklic  just shortening it to "From another device" and "From this app" which at least avoids some truncation/localisation issues.
